sdk/java-sdk-spring/src/it/java/com/example/wiring/pubsub/PublishBytesToTopic.java

```diff
@@ -0,0 +1,47 @@
+/*
+ * Copyright 2021 Lightbend Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.example.wiring.pubsub;
+
+
+import com.example.wiring.valueentities.customer.CustomerEntity;
+import com.fasterxml.jackson.core.JsonProcessingException;
+import kalix.javasdk.JsonSupport;
+import kalix.javasdk.action.Action;
+import kalix.javasdk.annotations.Publish;
+import kalix.javasdk.annotations.Subscribe;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.context.annotation.Profile;
+
+@Profile("docker-it-test")
+@Subscribe.ValueEntity(CustomerEntity.class)
+public class PublishBytesToTopic extends Action {
+
+  public static final String CUSTOMERS_BYTES_TOPIC = "customers_bytes";
+  private Logger logger = LoggerFactory.getLogger(getClass());
+
+  @Publish.Topic(CUSTOMERS_BYTES_TOPIC)
+  public Action.Effect<byte[]> handleChange(CustomerEntity.Customer customer) {
+    try {
+      var payload = JsonSupport.getObjectMapper().writerFor(CustomerEntity.Customer.class).writeValueAsBytes(customer);
```

Comment:
@octonato you suggested using `JsonSupport.decode` but then I would need to wrap bytes with Any, which I think makes no sense for this one.

Reply:
I see, but my comment was also misplaced. It is not `decode`, but `encode`. But doesn't matter. I see your point. How the user will take a payload and put it in bytes, it's their choice. Here we are just testing it.

Reply:
one more thought...

What happens if I subscriber to a topic that is emitting Protobuf (messages `Foo` and `Bar`)? 

I have the proto descriptor. I generate the message in my project and I create an Action with two methods receiving `Foo` and `Bar`. 

The Java SDK converts it to one method receiving `Any`, but it will expect it to be serialized to Json. The SDK will take the incoming Any payload and try to use the types `Foo.class` and `Bar.class` to deserialize it using the `JsonMessageCodec`.

We look already inside the Any to see if it is a json, but it's not, it fails. We need to be able to fallback to proto in that case.

Reply:
Just to sure that I understand your use case. One Kalix service produces proto messages to a topic. The second one, which is code-first, wants to consume from that topic. Am I correct? Looks like a separate issue to solve.

Reply:
Yes, more or less that. Don't even need to be another Kalix service. It suffices to have a Topic with proto and the consumer has the spec to generate the messages to consume. 

A Kalix user might expect it to work out of the box by adding methods receiving the proto messages directly.

Reply:
Now I'm confused. If the producer is not a Kalix service then we can't force to publish additional metadata like `ce-type` or `ce-datacontenttype` required for routing to a specific handler.

Reply:
Ok, so maybe I'm confusing something there. I thought that the proxy would be able to dispatch the messages based on the type, but that's won't work out to the box and the expected behavior in such a case is to receive the raw bytes and the user have to deal with it themselves.

Reply:
Yes, the proxy is able to do this based on additional metadata. Moreover, currently, we are passing only the metadata from the CloudEvent spec. For raw consumption, I would expect to get all the metadata from the topic. https://github.com/lightbend/kalix-jvm-sdk/issues/1024#issuecomment-1576327397